### PR TITLE
[FIX] On mobile Livechat covers the whole screen, making it intrusive for triggers

### LINKED ISSUE #244: On mobile Livechat covers the whole screen, making it intrusive for triggers.
If I setup a trigger for the livechat, say, with people that stayed 10 sec on the webpage, the chat window appears. Thought on desktop the window is not that obstructive, on mobile it is made full screen.

I think there should be an option to trigger popped down, such that the user can actually "accept" the interaction. Otherwise it is very intrusive.

image

Like this?

image

### PR DESCRIPTION
**Pull request by ashwaniYDV**
closes #244 #386

#### Description:
I've added a 5sec timeout trigger.

### CODE REVIEW

File: src/lib/triggers.js [hunk 1]
```diff
@@ -129,7 +137,10 @@ class Triggers {
 
 					route('/');
 					parentCall('openWidget');
-					store.setState({ minimized: false });
+
+					if (!window.mobilecheck()) {
```
Comment (renatobecker) [inline src/lib/triggers.js]: Instead of defining `minimized: false`, wouldn't it be better to skip calling the `parentCall` method?

Comment (ashwaniYDV) [reply src/lib/triggers.js]: That would be better (y)

But we have to skip both parentcall and setstate({minimized: false]) to not open widget in mobiles.

Comment (renatobecker) [reply src/lib/triggers.js]: By `skipping` I meant `return`...

File: src/lib/triggers.js [hunk 2]
```diff
@@ -9,6 +9,14 @@ import { normalizeAgent } from './api';
 
 const agentCacheExpiry = 3600000;
 let agentPromise;
+
+window.mobilecheck = function() {
```
Comment (renatobecker) [inline src/lib/triggers.js]: Please, move this code/function to the `helper` file.

Comment (ashwaniYDV) [reply src/lib/triggers.js]: By helpers did you mean helpers.stories.js?
Or should I create a new file `scr/helpers.js` as there is no helpers.js file

**Comment by ashwaniYDV** [comment]
@renatobecker Please review

File: src/lib/triggers.js (hunk 2 above)
Comment (ashwaniYDV) [reply src/lib/triggers.js]: done (y)

**Comment by dlecan** [comment]
It’s an interesting feature. Is it possible finish/merge it?

**Comment by dlecan** [comment]
@ashwaniYDV :+1: 

@renatobecker?

**Comment by patrick-tb** [comment]
Is there any news in this regard?

@ashwaniYDV 
@renatobecker

**Comment by renatobecker** [comment]
Fixed by -> https://github.com/RocketChat/Rocket.Chat.Livechat/pull/553

**Pull request closed without merging**